src/common/types.h

```diff
@@ -26,9 +26,11 @@ class CSmartPtr : public std::unique_ptr<T, void (*)(T *)> {
   CSmartPtr(T *object) : std::unique_ptr<T, void (*)(T *)>(object, deleter) {}
 };
 
-template <typename T, void (*deleter)(T *)> class DeleterType {
+template <typename T, void (*initializer)(T *), void (*deleter)(T *)>
+class InitializerDeleterWrapper {
```

Comment:
I'm not sure how to name this class better

Reply:
I'm not sure either. @lizan,do you have any ideas?

Reply:
perhaps `CSmartStruct` or `CSmartType`?